### PR TITLE
Migrate fetch call of task dependsOn to RTK query

### LINKED ISSUE #601: Migrate fetch call of task dependsOn to RTK query

### PR DESCRIPTION
**Pull request by vinit717**
### Issue:
close #601 

### Description:
Added a Hook to return  task title with its Id

### Dev Tested:
- [x] Yes

### Images/video of the change:
No changes as only migrate fetch call to RTK Query

### Follow-up Issues (if any)
No

### CODE REVIEW

File: __tests__/Unit/hooks/taskDetailsApi.test.tsx [hunk 1]
```diff
@@ -45,3 +45,25 @@ describe('useGetTaskDetailsQuery', () => {
...
+
+        await act(() => waitForNextUpdate());
+        const nextResponse = result.current;
+
+        expect(nextResponse.data).not.toBeUndefined();
+        expect(nextResponse.isLoading).toBe(false);
+        expect(nextResponse.isSuccess).toBe(true);
+        expect(nextResponse.data).toBeDefined();
+    });
+});
```
Comment (bharati-21) [inline __tests__/Unit/hooks/taskDetailsApi.test.tsx:69]: Please also test for the negative case - when there is a failure while fetching dependency details

File: src/components/taskDetails/index.tsx [hunk 2]
```diff
@@ -69,11 +67,14 @@ const TaskDetails: FC<Props> = ({ taskID }) => {
...
     const [isEditing, setIsEditing] = useState<boolean>(false);
-    const [taskTitle, setTaskTitle] = useState<string[]>([]);
-    const [id, setId] = useState<string[]>([]);
-    const [isFetched, setIsFetched] = useState<boolean>(false);
     const initialDataRef = useRef<Record<string, any> | undefined>({});
     const { data, isError, isLoading } = useGetTaskDetailsQuery(taskID);
+    const taskDependencyIds: string[] = data?.taskData?.dependsOn || [];
+    const {
+        data: dependencyData,
+        isLoading: Loading,
```
Comment (bharati-21) [inline src/components/taskDetails/index.tsx:75]: please avoid using capital case for constants

**Review by bharati-21 — CHANGES_REQUESTED** [review]

File: __tests__/Unit/hooks/taskDetailsApi.test.tsx (hunk 1 above)
Comment (vinit717) [reply __tests__/Unit/hooks/taskDetailsApi.test.tsx:99]: done

File: src/components/taskDetails/index.tsx (hunk 2 above)
Comment (vinit717) [reply src/components/taskDetails/index.tsx]: done

File: src/app/services/taskDetailsApi.ts [hunk 3]
```diff
@@ -1,14 +1,48 @@
...
+import { FetchBaseQueryError } from '@reduxjs/toolkit/dist/query';
+
+type dependency = PromiseSettledResult<{
+    title: string | undefined;
+    id: string;
+}>[];
 
 export const taskDetailsApi = api.injectEndpoints({
     endpoints: (build) => ({
         getTaskDetails: build.query<taskDetailsDataType, string>({
```
[the remark this answers is not in the document]
Comment (vinit717) [reply src/app/services/taskDetailsApi.ts:13]: This is not my change

File: __mocks__/handlers.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ import tagsHandler from './handlers/tags.handler';
 import levelsHandler from './handlers/levels.handler';
 import usersHandler from './handlers/users.handler';
 import taskTagsHandler from './handlers/taskTags.handler';
-import taskDetailsHandler from './handlers/task-details.handler';
+import {taskDetailsHandler} from './handlers/task-details.handler';
```
Comment (bharati-21) [inline __mocks__/handlers.ts:8]: Did you run prettier?

File: src/app/services/taskDetailsApi.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (bharati-21) [reply src/app/services/taskDetailsApi.ts:13]: > which element is of type string here

`taskId` -- query: (taskId): string => `${TASKS_URL}/${taskId}/details`

File: src/app/services/taskDetailsApi.ts [hunk 5]
```diff
@@ -1,14 +1,48 @@
...
+        getTasksDependencyDetails: build.query<dependency, string[]>({
+            queryFn: async (taskIds, _queryApi, _extraOptions, baseQuery) => {
+                try {
+                    const taskDetailsPromise = taskIds.map(async (taskId) => {
+                        const result = await baseQuery(
+                            `${TASKS_URL}/${taskId}/details`
+                        );
+
+                        if (result.error) {
+                            throw { error: result.error, id: taskId };
```
Comment (vvaibhavdesai) [inline src/app/services/taskDetailsApi.ts:26]: you are throwing the error from here but are you handling it at the parent level ?

File: src/components/taskDetails/index.tsx [hunk 6]
```diff
@@ -259,22 +229,38 @@ const TaskDetails: FC<Props> = ({ taskID }) => {
...
-                                                    onClick={() =>
-                                                        navigateToTask(
-                                                            id[index]
-                                                        )
-                                                    }
-                                                >
-                                                    {title}
-                                                </li>
-                                            </Link>
+                                    {loading || fetching ? (
```
Comment (vvaibhavdesai) [inline src/components/taskDetails/index.tsx:232]: this is not at all readable make this in to a proper if else

**Review by vvaibhavdesai — DISMISSED** [review]

File: src/app/services/taskDetailsApi.ts [hunk 7]
```diff
@@ -1,14 +1,48 @@
...
+                        return {
+                            title: task?.taskData?.title,
+                            id: taskId,
+                        };
+                    });
+                    const taskDetails = await Promise.allSettled(
+                        taskDetailsPromise
+                    );
+                    return { data: taskDetails };
+                } catch (error) {
```
[the remark this answers is not in the document]
Comment (vinit717) [reply src/app/services/taskDetailsApi.ts]: done

File: src/components/taskDetails/index.tsx (hunk 6 above)
Comment (vinit717) [reply src/components/taskDetails/index.tsx]: done

File: src/components/taskDetails/index.tsx [hunk 8]
```diff
@@ -251,26 +259,7 @@ const TaskDetails: FC<Props> = ({ taskID }) => {
...
-                                                    }
-                                                >
-                                                    {title}
-                                                </li>
-                                            </Link>
-                                        ))
-                                    ) : (
-                                        <p>No Dependency</p>
-                                    )}
+                                    {renderTaskDependencyList()}
```
Comment (bhtibrewal) [inline src/components/taskDetails/index.tsx:262]: suggestion : `<TaskDependencyList/>`

**Review by bhtibrewal — COMMENTED** [review]
left some comments, else looks good

File: src/components/taskDetails/index.tsx [hunk 9]
```diff
@@ -62,11 +57,21 @@ const TaskDetails: FC<Props> = ({ taskID }) => {
...
-    const [isFetched, setIsFetched] = useState<boolean>(false);
     const initialDataRef = useRef<Record<string, any> | undefined>({});
-    const { data, isError, isLoading } = useGetTaskDetailsQuery(taskID);
+    const { data, isError, isLoading, isFetching } =
+        useGetTaskDetailsQuery(taskID);
+
+    let taskDependencyIds: string[] = [];
+    if (!isFetching) {
+        taskDependencyIds = data?.taskData?.dependsOn || [];
+    }
```
Comment (bhtibrewal) [inline src/components/taskDetails/index.tsx:67]:
```suggestion
    const taskDependencyIds: string[] = !isFetching ? 
     data?.taskData?.dependsOn : []
```

**Review by bhtibrewal — DISMISSED** [review]
LGTM

**Review by bhtibrewal — APPROVED** [review]

**Review by bharati-21 — APPROVED** [review]
LGTM
